### PR TITLE
Pytorch Autologging Description Added To Tracking.rst 

### PR DESCRIPTION
**Pull request by karthik-77**
Signed-off-by: karthik-77 <karthiks@ideas2it.com>

## What changes are proposed in this pull request?

Pytorch autolog description was added to the tracking.rst file.

## How is this patch tested?

Documentation -NA

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/tracking.rst [hunk 1]
```diff
@@ -429,6 +429,41 @@ Autologging captures the following information:
...
 +-----------+------------------------+----------------------------------------------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 
+Pytorch (experimental)
+--------------------------
+
+Call :py:func:`mlflow.pytorch.autolog` before your training code to enable automatic logging of metrics and parameters. See example usages with `Pytorch <https://github.com/chauhang/mlflow/tree/master/examples/pytorch/MNIST>`_.
+
+In the current implementation, PyTorch autolog works with the `Lightning training loop <https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/training_loop.py>`_. The respective metrics associated with ``EarlyStopping Callabacks`` and ``pytorch_lightning.trainer`` are automatically logged. It must be noted that in case of a multi-optimizer scenario (such as usage of autoencoder) by default only the parameters for the first optimizer would be logged.
```
Comment (chauhang) [inline docs/source/tracking.rst:437]: @karthik-77 Please move the Optimizer note to the "Note" callout at the end (after the table)

File: docs/source/tracking.rst [hunk 2]
```diff
@@ -429,6 +429,41 @@ Autologging captures the following information:
...
+|                                          | ``restore_best_weight``, etc.                              | ``restore_best_weights``, etc                                                       |               |                                                                                                                                               |
++------------------------------------------+------------------------------------------------------------+-------------------------------------------------------------------------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------+
+| ``pytorch-lightning trainer``            |Training loss;validation loss;average_test_accuracy;        | ``fit()`` parameters; optimizer name; learning rate; epsilon                        | --            | Model summary on training start; `MLflow Model <https://mlflow.org/docs/latest/models.html>`_ (Pytorch model) on training end;                |
```
Comment (chauhang) [inline docs/source/tracking.rst:451]: @karthik-77 Please change the sequence of the table rows. Move the `pytorch-lightning trainer` row to be the 1st row in the table

**Review by chauhang — CHANGES_REQUESTED** [review]
@karthik-77 Please see the comments inline. Rest look good

File: docs/source/tracking.rst (hunk 1 above)
Comment (karthik-77) [reply docs/source/tracking.rst]: Yes i have moved the Optimizer  note to the "Note" Callout at the end

File: docs/source/tracking.rst (hunk 2 above)
Comment (karthik-77) [reply docs/source/tracking.rst]: I have changed the order. First row is the pytorch lightning trainer and the callback comes next,

File: docs/source/tracking.rst [hunk 3]
```diff
@@ -429,6 +429,44 @@ Autologging captures the following information:
...
+|                                          |                                                            |                                                                                     |               |                                                                                                                                               |
++------------------------------------------+------------------------------------------------------------+-------------------------------------------------------------------------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------+
+|``trainer.callbacks.earlystopping``       |Training loss;validation loss;average_test_accuracy;        | ``fit()`` parameters; optimizer name; learning rate; epsilon                        | --            | Model summary on training start; `MLflow Model <https://mlflow.org/docs/latest/models.html>`_ (Pytorch model) on training end;                |
```
Comment (smurching) [inline docs/source/tracking.rst:449]: Should this be pytorch_lightning.callbacks.early_stopping? https://pytorch-lightning.readthedocs.io/en/stable/early_stopping.html#early-stopping-based-on-metric-using-the-earlystopping-callback

I don't think a `trainer.callbacks.earlystopping` module exists

File: docs/source/tracking.rst [hunk 4]
```diff
@@ -429,6 +429,44 @@ Autologging captures the following information:
...
+| Framework/module                         | Metrics                                                    | Parameters                                                                          | Tags          | Artifacts                                                                                                                                     |
++------------------------------------------+------------------------------------------------------------+-------------------------------------------------------------------------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------+
+|``pytorch-lightning trainer``             | Training loss; validation loss;average_test_accuracy;      | ``fit()`` parameters; optimizer name; learning rate; epsilon.                       | --            | Model summary on training start; `MLflow Model <https://mlflow.org/docs/latest/models.html>`_ (Pytorch model) on training end;                |
```
Comment (smurching) [inline docs/source/tracking.rst:443]:
```suggestion
|Pytorch-Lightning Trainer``             | Training loss; validation loss;average_test_accuracy;      | ``fit()`` parameters; optimizer name; learning rate; epsilon.                       | --            | Model summary on training start; `MLflow Model <https://mlflow.org/docs/latest/models.html>`_ (Pytorch model) on training end;                |
```

**Review by smurching — COMMENTED** [review]
Pushed some edits & had a few suggestions, otherwise looks good thanks @karthik-77!

**Comment by karthik-77** [comment]
Hi Siddarth,

Thanks for pointing that out. I will work on the suggestion and update the
file.

Thanks and regards
karthik

**Comment by karthik-77** [comment]
@smurching request you to kindly sign off the commits.

**Review by chauhang — APPROVED** [review]

File: docs/source/tracking.rst (hunk 3 above)
Comment (karthik-77) [reply docs/source/tracking.rst]: Yes. The change has been incorporated.

File: docs/source/tracking.rst (hunk 4 above)
Comment (karthik-77) [reply docs/source/tracking.rst]: The suggested change has been applied.

**Review by harupy — APPROVED** [review]
LGTM
